src/components/Ad.js

```diff
@@ -0,0 +1,39 @@
+import React, { Component } from "react";
+import { string, oneOfType, arrayOf, node } from "prop-types";
+import { Helmet } from "react-helmet";
+
+import pick from "../utils/pick";
+import commonPropTypes from "./commonPropTypes";
+
+class Ad extends Component {
+  static propTypes = {
+    ...commonPropTypes,
+    type: string,
+    "data-slot": string.isRequired,
+    children: oneOfType([arrayOf(node), node])
+  };
+
+  static defaultProps = {
+    children: null
+  };
+
+  get properties() {
+    return pick(this.props, ...[...Object.keys(commonPropTypes)]);
```

Comment:
am I wrong or here does not makes much sense to descructure, restructure and descructure
looks like we can write it like
```
return pick(this.props, ...Object.keys(commonPropTypes));
```

Reply:
this is a copy of a method used in amp-sidebar. I assumed it was ok for you if it was in the code

Reply:
not exactly ...in the sidebar 'id' and 'side' are added ... even though ugly.. still makes some sense :D

Reply:
done

Reply:
👍